### ISSUE TITLE
Getting NullPointerException when getting allocation from ApplicationServer

### ISSUE THREAD

**Issue by LooFifteen** [mattmalec/Pterodactyl4J#6]
Hey. I'm receiving a NullPointer when I do
```java
Allocation allocation = server.retrieveAllocation().execute();
String ip = allocation.getIP() + ":" + allocation.getPort(); // <-- Error Here
```
I'm not sure why this is happening, I think its only happening with some servers, most are working, this and a few others aren't.
I wouldn't think that not using Async would be a cause? If you could help that would be great!

**Comment by mattmalec** [comment]
What version of Pterodactyl do you happen to be using?

**Comment by LooFifteen** [comment]
v0.7.17, and I forgot to mention, I believe that other methods are working, just not the Allocations.

**Comment by mattmalec** [comment]
Weird, I just tried it on my end and it worked fine. Also, `allocation.getIP() + ":" + allocation.getPort();` can be replaced with `allocation.getFullAddress();`

**Comment by LooFifteen** [comment]
Yeah I replaced it with that because I was trying to see if getFullAddress wasn't working. This is quite odd, it works for some servers and not with others. Would there be any reason not to give an allocation?

**Comment by mattmalec** [comment]
Mind sharing the full stacktrace?

**Comment by LooFifteen** [comment]
The stacktrace isn't very helpful but:
```
java.lang.NullPointerException
at dev.sllcoding.mineryze.mineryzeservermanagement.Utils.PteroUtils$Pterodactyl.getAddress(PteroUtils.java:61)
at dev.sllcoding.mineryze.mineryzeservermanagement.Utils.ServerUtils.connect(ServerUtils.java:25)
at dev.sllcoding.mineryze.mineryzeservermanagement.Utils.PteroUtils$Pterodactyl.join(PteroUtils.java:72)
at dev.sllcoding.mineryze.mineryzeservermanagement.Commands.JoinCommand.execute(JoinCommand.java:32)
at net.md_5.bungee.api.plugin.PluginManager.dispatchCommand(PluginManager.java:192)
at net.md_5.bungee.api.plugin.PluginManager.dispatchCommand(PluginManager.java:142)
at net.md_5.bungee.connection.UpstreamBridge.handle(UpstreamBridge.java:149)
at net.md_5.bungee.protocol.packet.Chat.handle(Chat.java:66)
at net.md_5.bungee.netty.HandlerBoss.channelRead(HandlerBoss.java:105)
```
I'm not using async if you think thats the issue?

**Comment by mattmalec** [comment]
Async/sync shouldn't matter in this case. What's on `PteroUtils` line 61?

**Comment by LooFifteen** [comment]
I mentioned it in my original message:
```java
Allocation allocation = server.retrieveAllocation().execute();
String ip = allocation.getIP() + ":" + allocation.getPort(); // <-- Error Here
```

**Comment by mattmalec** [comment]
Those are two lines - I'm looking for one.

**Comment by LooFifteen** [comment]
Line 61 is `// <-- Error Here`

**Comment by mattmalec** [comment]
So, all Allocation methods don't work? What about retrieving another field from that same server? I'm beginning to think the Server itself is null.

**Comment by LooFifteen** [comment]
I'm loading up my IDE now, I think its just allocations, I'll check. This might take me a few minutes so...

Edit: I think its just allocations because I'm getting the server's name perfectly fine.

Edit 2: Yeah so I think its just Allocations because I run this just a few methods earlier, which works:
```java
player.sendMessage(new TextComponent(server.getName() + " is online, sending you there now."));
ServerUtils.connect(player, server);
```

**Comment by mattmalec** [comment]
Can you try to retrieve something? Like the location or node?

**Comment by LooFifteen** [comment]
![image](https://user-images.githubusercontent.com/39528861/89009976-9ff2ad00-d305-11ea-9407-a6f9d3012b6c.png)
1st line -> Name
2nd line -> Node
3rd line -> Error because of Null Allocation

**Comment by mattmalec** [comment]
Try and output the executed response of `ApplicationServer#retrieveAllocations`. It should be a list of all Allocations for all of your nodes.

**Comment by LooFifteen** [comment]
Do you mean PteroApplication#retrieveAllocations?

**Comment by mattmalec** [comment]
Oops, yes I meant `PteroApplication#retrieveAllocations`.

**Comment by mattmalec** [comment]
Have you also tried printing `Allocation#toString`? It should be the raw JSON response from the API.

**Comment by LooFifteen** [comment]
I'll try both right now.

**Comment by LooFifteen** [comment]
Ok so retrieveAllocations is definitely returning all of the panel's allocations.

**Comment by mattmalec** [comment]
Did you happen to find the allocation for your server included in that list?

**Comment by LooFifteen** [comment]
It was spamming so ptero killed the instance of where it was outputting the data, so, no.

**Comment by mattmalec** [comment]
Ah, and the `Allocation#toString`?

**Comment by LooFifteen** [comment]
Oh yeah, forgot about that, just a sec.

Edit: NullPointer when I Allocation#toString

**Comment by mattmalec** [comment]
I wonder if you've got something weird going on with your panel specifically. I've just tried two other panels alongside mine and the Allocation methods are all working as expected.

**Comment by LooFifteen** [comment]
I've `toString`ed the server, it has `"allocation": 1` and `"allocations": 0`. Would this be the cause of it?

Edit: Just compared to a server that works, I don't believe it's the issue.

**Comment by mattmalec** [comment]
So one server works but the other doesn't?

**Comment by LooFifteen** [comment]
Yes

**Comment by mattmalec** [comment]
Given that allocations do in fact work except on that one server, it's pretty safe to say that would unfortunately be an issue on your end, and nothing to do with my wrapper. I don't believe it has anything to do with your code, rather, your panel.

**Comment by LooFifteen** [comment]
Okay, thanks for the help anyway!

**Comment by mattmalec** [comment]
I just looked into this again and I think I found something internal causing it. How many allocations do you happen to have on your panel?

**Comment by LooFifteen** [comment]
I have about 2000 allocations

**Comment by mattmalec** [comment]
That might be why. Just released a new 1.1 version - feel free to update to that version now. It fixes the NPE with the allocations and a number of other paginated elements.
